### ISSUE TITLE
Conditional Logic on Headers

### ISSUE THREAD

**Issue by wgstjf** [rilwis/meta-box#682]
Hi there, quick one for you...does Conditional Logic work on Heading fields?

Cheers,

Will

**Comment by tanng** [comment]
Yes, it works on heading field :) This is an example: only show heading when post format is "aside".

```
	$meta_boxes[] = [
		'id' 		=> 'foo',
		'title' 	=> 'Foo',
		'fields' 	=> [
			[
				'id' 	=> 'heading',
				'name'	=> 'Heading',
				'type' 	=> 'heading',
				'std' 	=> 'Heading',
				'visible' => ['post_format', 'aside']
			]
		]
	];
```

**Comment by wgstjf** [comment]
Thanks for the quick reply.

It doesn't seem to be working for me.  I am complicating the issue by having the header within a cloneable group as well.  The image shows what I have I have so far.  The idea here is that the 'Additional Options' heading should only show if the 'Static Content Panel' select list has 'Testimonials Bar' selected.

You can see from the screenshot that this is working fine for the row which has 'Filter Testimonials' in it as this row only shows when the 'Testimonials Bar' is selected.  Using the exact same rule that controls those on the 'Heading' field has no effect.  As you can see the 'Additional Options' heading shows up whether or not 'Testimonials Bar' is selected.

![screenshot](https://cloud.githubusercontent.com/assets/3740694/10817083/8aa005d6-7e2f-11e5-9b53-cddcbfb7b973.JPG)

The following is an extract from my code:

```php
  // HEADING
            array(
              'type' => 'heading',
              'name' => 'Additional Options',
              'id'   => 'heading_additional_options', // Not used but needed for plugin
              'columns'   => '12',
              'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),
            ),
            
            // SELECT -Filter Testimonials
            array(
                'name'             => 'Filter testimonials',
                'id'               => $prefix."filter_testimonials",
                'class'               => $prefix."filter_testimonials",
                'type'             => 'select',
                'options'          => array (
                    'no'    =>  'No',
                    'yes-service'    =>  'Yes by Service',
                    'yes-product'    =>  'Yes by Product',
                ),
                'desc' => 'Do you want to filter which testimonials are shown?',
                'std'              => '',
                'clone'              => false,
                'columns'  => '4',
                'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),
            ),
```

Thanks,

Will

**Comment by tanng** [comment]
Hi Will,

Can you please check if another field id similar to `heading_additional_options` (For example: options or 'additional' or something else), or please change `heading_additional_options` to some other dummy ID, for example: `f014raa_opt`. If it still not work, can you please provide me full code of your meta box.

Thanks

Tan

**Comment by wgstjf** [comment]
Tan,

Thanks for getting on this.  I have tried with your dummy ID and there's no change.  Would it be anything to do with the fact that a Heading does not get assigned a 'name' attribute?  A quick look at the show/hide code seems to suggest that it relies on the name rather than the 'id' to target a grouped and cloneable element?

If the answer is no then I can provide the code but there's quite a lot of it :-)

Cheers,

Will

**Comment by tanng** [comment]
Hi Will,

Conditional Logic uses ID to detect fields. When you look into the code, you see the name and that name generated by field ID. So basically, you don't have to care about name, just ID.

If the code is too long, can you create a Gist and I can copy it, or better give me FTP/Credentials to your website if available so I can check quicker.

Cheers.

**Comment by wgstjf** [comment]
Thanks Tan,

Here's the code.  As far as I can see the Heading is not being targeted by the conditional logic script at all.  The data-visible tag or data-hidden is not being applied and the only difference I can see, at a glance, is the that 'id' of the heading element is not having an instance number appended to it. 

```php
	//Page Metaboxes
	$prefix = 'static_';
	$meta_boxes[] = array(
		'id'       => 'static',
		'title'    => 'Static Content Setup',
		'pages'    => array( 'page'),
		'context'  => 'normal',
	  'priority' => 'low',
		'fields' => array(

      // HEADING
      array(
        'type' => 'heading',
        'name' => 'Define your page static content here. Clone as necessary',
        'id'   => 'fake_id', // Not used but needed for plugin
        'columns'   => '12',
      ),
      // Group
      array(
          //'name' => 'Static Content Bars:', // Optional
          'type' => 'group',
          'id' =>  $prefix .'content_',
          'class' =>  '',
          'clone' => true,
          'sort_clone' => true,
          'columns'     => '12',
          // List of sub-fields
          'fields' => array(
              // SELECT - Static Content Panels
        			array(
            			'name'             => 'Static Content Panel',
            			'id'               => $prefix."shortcodes",
            			'class'               => $prefix."shortcodes",
            			'type'             => 'select',
            			'options'          => array (
                      ''    =>  'None',
                      'stats-bar'    =>  'Stats Bar',
                      'trending-bar'    =>  'Trending Bar',
                      'testimonials-bar'    =>  'Testimonials Bar',
                      'services-bar'    =>  'Services List Bar',
                      'products-bar'    =>  'Products List Bar',
                      'offers-bar'    =>  'Featured Offer Bar',
                      'partners-bar'    =>  'Partner Logos Bar',
                      'contact-bar'    =>  'Contact/Callback Bar',
                  ),
            			'desc' => 'Select item(s) to show under the main content.  Clone as necessary to build page.',
                  'std'              => '',
                  'clone'              => false,
                  'columns'  => '4',
        			),
              // SELECT - Position Bar
        			array(
            			'name'             => 'Position Bar',
            			'id'               => $prefix."position",
            			'class'               => $prefix."position",
            			'type'             => 'select',
            			'options'          => array (
                      'below'    =>  'Below Secondary Content',
                      'above'    =>  'Above Secondary Content',
                  ),
            			'desc' => 'Position bar relative to secondary content panel.',
                  'std'              => '',
                  'clone'              => false,
                  'columns'  => '4',
        			),
                // RADIO - Background Colour
        			array(
        					'name'             => 'Bar Background Colour Override',
        					'id'               => $prefix."colour",
        					'class'            => $prefix."colour header_colour_picker",
        					'type'             => 'radio',
        					'options'          => $bar_colours_menu,
        					'desc' => 'Change the default background colour of this bar?',
                  'std'              => 'default',
                  'columns'  => '4',

        			),

            // HEADING
            array(
              'type' => 'heading',
              'name' => 'Additional Options',
              'id'   => $prefix.'f014raa_opt', // Not used but needed for plugin
              'columns'   => '12',
              'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),
            ),

            // SELECT -Filter Testimonials
            array(
                'name'             => 'Filter testimonials',
                'id'               => $prefix."filter_testimonials",
                'class'               => $prefix."filter_testimonials",
                'type'             => 'select',
                'options'          => array (
                    'no'    =>  'No',
                    'yes-service'    =>  'Yes by Service',
                    'yes-product'    =>  'Yes by Product',
                ),
                'desc' => 'Do you want to filter which testimonials are shown?',
                'std'              => '',
                'clone'              => false,
                'columns'  => '4',
                'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),
            ),
            // SELECT - Related Service
              array(
                  'name'             => 'Related Service',
                  'id'               => $prefix."related_service",
                  'class'            => "",
                  'type'             => 'select',
                  'options'          => $service_categories_menu,
                  'desc' => 'If applicable filter testimonials chosen by related Service',
                  'std'              => 'sw_blue',
                  'columns'  => '4',
                  'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),

              ),
            // SELECT - Related Product
              array(
                  'name'             => 'Related Product',
                  'id'               => $prefix."related_product",
                  'class'            => "",
                  'type'             => 'select',
                  'options'          => $product_categories_menu,
                  'desc' => 'If applicable filter testimonials chosen by related Service',
                  'std'              => 'sw_blue',
                  'columns'  => '4',
                  'hidden' => array( $prefix.'shortcodes', 'not in', array('testimonials-bar') ),

              ),
    			),
			),
    ),
  );
```

Any help greatly appreciated.

Will

**Comment by tanng** [comment]
lol my fault, when a field in Group, Conditional Logic interact with that field by [name] instead of [id]. Heading field doesn't contains ID so it doesn't work for now, sorry. We'll fix this bug on the next release. Sorry about this.

Regards
Tan

**Comment by wgstjf** [comment]
No need for apologies, just glad it wasn't something I had missed! :-)

Any idea when the next release might be as I am pushing hard to get this project completed?  If it is going to be a while then I'll just remove the heading.

Cheers,

Will

**Comment by tanng** [comment]
I'm currently try my best to finish the big update for Meta Box Builder, so the update for Conditional Logic will on Nov, 20th.

**Comment by tanng** [comment]
Hi Will,

I have just released v1.2 which solved this problem. Please upgrade.

Cheers.
